File: php/features.php
<?php
declare(strict_types=1);

// CatchdomsSecurity SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CatchdomsSecurityFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CatchdomsSecurityBaseFeature();
            case "test":
                return new CatchdomsSecurityTestFeature();
            default:
                return new CatchdomsSecurityBaseFeature();
        }
    }
}

File: py/catchdomssecurity_sdk/features.py
# CatchdomsSecurity SDK feature factory

from catchdomssecurity_sdk.feature.base_feature import CatchdomsSecurityBaseFeature
from catchdomssecurity_sdk.feature.test_feature import CatchdomsSecurityTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CatchdomsSecurityBaseFeature(),
        "test": lambda: CatchdomsSecurityTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
